### PR TITLE
Re-align tsconfig with next recommendations

### PR DESCRIPTION
**Pull request by spolu**
## Description

Re-align tsconfig to the recommendations of the next boilerplate.

Test:

- test front-edge
- test upsert async on front-edge

## Risk

Minimal once tested on front-edge

## Deploy Plan

- deploy `front`

### CODE REVIEW

File: front/tsconfig.json [hunk 1]
```diff
@@ -9,12 +8,16 @@
     "noEmit": true,
     "esModuleInterop": true,
     "module": "esnext",
-    "moduleResolution": "node",
+    "moduleResolution": "bundler",
```
Comment (flvndvd) [inline front/tsconfig.json]: IMHO we should probably go with `nodenext` so we can keep a broader compatibility and one day maybe have .js.map 😅.

File: front/tsconfig.json [hunk 2]
```diff
@@ -9,12 +8,16 @@
...
     "resolveJsonModule": true,
     "isolatedModules": true,
     "jsx": "preserve",
     "incremental": true,
-    "downlevelIteration": true,
```
Comment (flvndvd) [inline front/tsconfig.json:17]: Seem safe to remove this as we remove the target: "es5"

Comment (flvndvd) [reply front/tsconfig.json:17]: See recommendations here: https://www.npmjs.com/package/@tsconfig/node20

File: front/tsconfig.json (hunk 1 above)
Comment (spolu) [reply front/tsconfig.json]: done

File: front/pages/api/auth/[auth0].ts [hunk 3]
```diff
@@ -6,24 +6,24 @@ import {
...
 import type { NextApiRequest, NextApiResponse } from "next";
 
 import config from "@app/lib/api/config";
 
 export default handleAuth({
   login: handleLogin((req) => {
-    const connection = "query" in req ? req.query.connection : undefined;
+    let connection: string | undefined = undefined;
+    if ("query" in req && req.query.connection) {
+      connection = Array.isArray(req.query.connection)
```
Comment (flvndvd) [inline front/pages/api/auth/[auth0].ts:17]: When is this happening?

Comment (spolu) [reply front/pages/api/auth/[auth0].ts:17]: req.query can be a string as per typescript you can do req[]=1&req[]=2 I believe

Comment (flvndvd) [reply front/pages/api/auth/[auth0].ts:17]: I see, common pattern we use in the code is if `!== string` we return an error. In this case you can redirect to `      return res.redirect("/login-error");`.

**Review by flvndvd — APPROVED** [review]
LGTM

**Comment by spolu** [comment]
Deploying front-edge

**Comment by spolu** [comment]
Everything looks good. Moving forward.

**Pull request merged**